### PR TITLE
Custom Report Fix In Scope Only Option (Issue 2492)

### PR DESCRIPTION
**Pull request by kingthorin**
ReportParser > Removed. Functionality added to ReportLastScan.
ReportLastScan > Modified to handle the "In Scope Only Option" and "Selected Alerts".
build.xml > Add `deploy-customreport-without-help-indexes` target.
ZapAddOn.xml > Updated changes section.
.project > Removed (not needed, not maintained).

Fixes zaproxy/zaproxy#2492

### CODE REVIEW

**Comment by kingthorin** [comment]
The indentation in this file is mixed, I tried to be consistent but it was really awkward for this code block.

There seems to be a `.project` file in this extension directory which doesn't seem to be necessary or maintained, any objection to its removal being included in this PR (clean code and what not :+1: ).

**Comment by thc202** [comment]
No.

**Comment by kingthorin** [comment]
Thanks, done.

File: src/org/zaproxy/zap/extension/customreport/ReportParser.java [hunk 1]
```diff
@@ -51,10 +51,22 @@ public static StringBuilder deleteNotInScope( List<Context> contexts, StringBuil
 	    	   // check if included in scope
 	    	   for( int i = 0 ; i < contexts.size(); i++){
 	    		   Context context = contexts.get(i);
-	    		   String url = site.attributeValue("name").toString();
-	    		   if ( !context.isIncluded(url) ){
-	    			   sites.remove(site);
-	    		   }
+
+		           @SuppressWarnings("unchecked")
+		           List<Element> alerts = site.elements("alerts");
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/customreport/ReportParser.java]: There's just one element `alerts` (per site), it could use `element` instead.

**Comment by thc202** [comment]
If you have some spare cycles, I'd suggest clean up a little more, remove `ReportParser` and do the filtering in `ReportLastScan` (before adding the alerts to the report).

**Comment by kingthorin** [comment]
Will do

**Comment by thc202** [comment]
Thanks.

**Comment by kingthorin** [comment]
Done

**Comment by thc202** [comment]
Thank you!

**Pull request merged**